### PR TITLE
Devel monotonic

### PR DESCRIPTION
**Pull request by iancoolidge**
For issue https://github.com/grpc/grpc/issues/6824

I'm hoping to replace all use of realtime clock space with monotonic inside gRPC,
and provide API amendments that facilitate that in C++.

I don't believe that this branch is complete yet, but am hoping for some early feedback.

### CODE REVIEW

**Comment by ctiller** [comment]
I'm seeing a bunch of Python changes here too: any particular reason?

File: src/core/lib/support/sync_posix.c [hunk 1]
```diff
@@ -69,7 +69,12 @@ int gpr_mu_trylock(gpr_mu* mu) {
 
 /*----------------------------------------*/
 
-void gpr_cv_init(gpr_cv* cv) { GPR_ASSERT(pthread_cond_init(cv, NULL) == 0); }
+void gpr_cv_init(gpr_cv* cv) {
+  pthread_condattr_t attributes;
+  GPR_ASSERT(pthread_condattr_init(&attributes) == 0);
+  GPR_ASSERT(pthread_condattr_setclock(&attributes, CLOCK_MONOTONIC) == 0);
```
Comment (ctiller) [inline src/core/lib/support/sync_posix.c]: This isn't going to work on mac... suggest bifurcating this file into sync_posix.c, sync_linux.c and making the monotonic changes only for sync_linux.c.

**Comment by nicolasnoble** [comment]
Is this one stale ?

**Comment by ctiller** [comment]
Ping. Any interest in finishing this off? I plan to close the request next week if not.

**Comment by barrierbank** [comment]
This would be a great improvement because at the moment, if you change your system clock, gRPC's timeouts break.

Is there a successor to this PR?

**Comment by vivek-avi** [comment]
facing the same issue with system_clock changes due to NTP drift, need to use steady_clock instead, are there any plans to resurrect this / other alternatives ?

**Comment by srini100** [comment]
@vivek-avi, please open a new issue if one doesn't already exist. It will be easier to track than commenting on an old closed PR. Thanks.

**Comment by vivek-avi** [comment]
> @vivek-avi, please open a new issue if one doesn't already exist. It will be easier to track than commenting on an old closed PR. Thanks.

Thanks @srini100, opened a new issue for tracking
https://github.com/grpc/grpc/issues/16891

**Pull request closed without merging**